src/util/apiUtlis.ts

```diff
@@ -14,30 +14,36 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
 
-interface DispatchGenericCallInput {
-  call: Function;
-  dispatch?: (error: string | null, data: any, loading: boolean) => void;
-  emptyData?: unknown;
-}
+import logger from '../util/logger';
 
-interface Result {
+interface Result<T> {
   error: string | null;
-  data: unknown | null;
+  data: T | null;
+}
+interface DispatchGenericCallInput<T> {
+  call: (...args: any[]) => Promise<T>;
```

Comment:
```suggestion
  call: () => Promise<T>;
```

Reply:
I wouldn't make the change ... If for whatever reason we want to provide a parameter to the useCallback function this will complain. My original code handles both cases.

Reply:
No it does not.